compiler/rustc_middle/src/query/mod.rs

```diff
@@ -846,8 +846,10 @@ rustc_queries! {
         separate_provide_extern
     }
 
-    query issue33140_self_ty(key: DefId) -> Option<ty::EarlyBinder<ty::Ty<'tcx>>> {
-        desc { |tcx| "computing Self type wrt issue #33140 `{}`", tcx.def_path_str(key) }
+    query self_ty_of_trait_impl_enabling_order_dep_trait_object_exploit(
```

Comment:
```suggestion
    query self_ty_of_trait_impl_enabling_order_dep_trait_object_hack(
```
I think exploit is not a good fithere 🤔